sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -88,14 +89,70 @@ case class AnalyzeTableCommand(tableName: String) extends RunnableCommand {
             }
           }.getOrElse(0L)
 
-        // Update the Hive metastore if the total size of the table is different than the size
-        // recorded in the Hive metastore.
-        // This logic is based on org.apache.hadoop.hive.ql.exec.StatsTask.aggregateStats().
-        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+        var needUpdate = false
+        val totalSize = if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+          needUpdate = true
+          newTotalSize
+        } else {
+          oldTotalSize
+        }
+        var numRows: Option[BigInt] = None
+        if (!noscan) {
+          val oldRowCount: Long = if (catalogTable.catalogStats.isDefined) {
```

Comment:
`catalogTable.catalogStats.flatMap(_.rowCount.map(_.toLong)).getOrElse(-1L)` is shorter. It would also a bit shorted if you put `catalogStats` in its own val.